### PR TITLE
add otel agent integration test to gitlab ci

### PR DESCRIPTION
**Pull request by jackgopack4**
### What does this PR do?
Adds the existing otel agent integration test to the gitlab pipeline

### Motivation
This test is not run on CI, see https://github.com/DataDog/datadog-agent/pull/31437 where I set this test to assert 0=1 and all pipelines passed

### Describe how to test/QA your changes
`inv otel-agent.integration-test`
### Possible Drawbacks / Trade-offs
more overhead on pipelines
### Additional Notes
Job passed on CI: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/719768933

### CODE REVIEW

File: tasks/otel_agent.py [hunk 1]
```diff
@@ -69,3 +69,15 @@ def image_build(ctx, arch='amd64', base_version='latest', tag=OT_AGENT_TAG, push
...
     os.remove(os.path.join(build_context, BIN_NAME))
     os.remove(os.path.join(build_context, CFG_NAME))
+
+
+@task
+def integration_test(ctx):
+    """
+    Run the otel integration test
+    """
+    cmd = """go test -timeout 0s -tags kubelet,jmx,cri,otlp,zstd,zk,jetson,etcd,\
```
Comment (ankitpatel96) [inline tasks/otel_agent.py:79]: are you sure we need all these tags?  don't we just need -otlp and maybe -test?

Comment (jackgopack4) [reply tasks/otel_agent.py:79]: probably fine to do that instead

**Comment by jackgopack4** [comment]
how on earth this is a "medium review" smh

**Review by CelianR — APPROVED** [review]

**Review by ankitpatel96 — APPROVED** [review]

**Review by KevinFairise2 — APPROVED** [review]

**Pull request merged**